src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -229,4 +229,69 @@ int getNumberOfCPUCores(bool bigCoresOnly) {
     return phys_cores;
 }
 
+bool cpuMapAvailable() {
+    return cpu._cpu_mapping_table.size() > 0;
+}
+
+int getCoreOffset(const cpu_core_type_of_processor core_type) {
+    int offset = 0;
+    if (core_type <= EFFICIENT_CORE_PROC && core_type >= ALL_PROC) {
+        for (int i = 0; i < cpu._processors; i++) {
+            if (cpu._cpu_mapping_table[i][CPU_MAP_CORE_TYPE] == core_type &&
+                cpu._cpu_mapping_table[i][CPU_MAP_USED_FLAG] <= 0) {
+                offset = i;
+                break;
+            }
+        }
+    } else {
+        IE_THROW() << "Wrong value for core_type " << core_type;
+    }
+    return offset;
+}
+
+int getThreadStep(const cpu_core_type_of_processor core_type) {
+    std::vector<int> proc_array;
+    if (core_type <= EFFICIENT_CORE_PROC && core_type >= ALL_PROC) {
+        for (int i = 0; i < cpu._processors; i++) {
+            if (cpu._cpu_mapping_table[i][CPU_MAP_CORE_TYPE] == core_type &&
+                cpu._cpu_mapping_table[i][CPU_MAP_USED_FLAG] <= 0) {
+                proc_array.push_back(i);
+            }
+            if (proc_array.size() == 2) {
+                break;
+            }
+        }
+    } else {
+        IE_THROW() << "Wrong value for core_type " << core_type;
+    }
+    return proc_array.size() == 2 ? proc_array[1] - proc_array[0] : 1;
+}
+
+std::vector<int> getAvailableCPUs(const cpu_core_type_of_processor core_type, const int num_cpus) {
```

Comment:
For stream of logic core, need function to get paired processors of stream of physical core.